Web Quiz Engine (Kotlin)/task/src/engine/service/QuizService.kt

```diff
@@ -9,31 +9,44 @@ import engine.entity.Quiz
 import org.springframework.stereotype.Service
 
 @Service
-class QuizService(private val quizRepository: WebQuizRepository) {
+class QuizService(
+    private val quizRepository: WebQuizRepository,
+) {
 
     fun findQuizById(
         id: Int,
         omitId: Boolean = false,
         omitAnswerList: Boolean = false
         ) =
-        quizRepository.getQuizById(
-            id = id,
-            omitId = omitId,
-            omitAnswerList = omitAnswerList,
-    )?: throw QuizNotFoundException(quizId = id)
+        quizRepository
+            .findById(id)
+            .orElseThrow { QuizNotFoundException(quizId = id) }
+            .let { quiz ->
+                quiz.copy(
+                id = if (omitId) null else id,
```

Comment:
Using Booleans like this feels kinda odd to me.
Looking at your code in general, I see that you are returning the Quiz dabatase entity in your Controllers for some of the endpoints, so now I understand why you re using Booleans like this.

The cleaner way of doing this would be to define a `QuizDto` (chose the naming you prefer) object in your `model` package. This class would be dedicated to be mapped from the `Quiz` class, and when you would map it, you would build it the way that you need it to be according to your Controller's contract and business rules.

I understand however that this is kinda heavy for this kind of small setup, but it would be better infrastructure.

Reply:
It is kinda heavy but I'm here to learn. It also is something I could have known better. I think I will do as you say and have extension functions to go from one to the other. I'll report back here.

Reply:
So I ended up -per your advice- implementing a new `QuizDto` class annotated with `@JsonInclude(JsonInclude.Include.NON_NULL)` and map from one to the other in the controller
Done